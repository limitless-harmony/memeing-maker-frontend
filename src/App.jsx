import React from 'react';
import PropTypes from 'prop-types';
import { ConnectedRouter } from 'connected-react-router';
import styled from 'styled-components';

import routes from 'routes';
import { calculateRem, mobileWidth } from 'styles';

export const App = ({ history }) => {
  return (
    <ConnectedRouter history={history}>
      <MainContainer>{routes}</MainContainer>
    </ConnectedRouter>
  );
};

const MainContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  font-size: inherit;
  padding: 0 ${calculateRem(14)};
  width: 100%;
  max-width: ${mobileWidth};
`;

App.propTypes = {
  history: PropTypes.shape({}).isRequired,
};

export default App;
