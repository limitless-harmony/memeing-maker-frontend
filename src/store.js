import { createBrowserHistory } from 'history';
import { applyMiddleware, compose, createStore } from 'redux';
import { routerMiddleware } from 'connected-react-router';

import rootReducer from 'reducers';

export const history = createBrowserHistory();

const configureStore = preLoadedState => {
  const composeEnhancer =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    rootReducer(history),
    preLoadedState,
    composeEnhancer(applyMiddleware(routerMiddleware(history)))
  );

  return store;
};

export default configureStore;
