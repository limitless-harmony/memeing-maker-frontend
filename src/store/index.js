import { createBrowserHistory } from 'history';
import { applyMiddleware, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router';
import { throttle } from 'lodash';

import { saveState, loadState } from 'services/persistState';
import rootReducer from 'reducers';

export const history = createBrowserHistory();
const middlewares = [thunk, routerMiddleware(history)];
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const preLoadedState = loadState();

// Middlewarees only in development
if (process.env.NODE_ENV === 'development') {
  // eslint-disable-next-line global-require
  const { logger } = require('redux-logger');
  middlewares.push(logger);
}

const enhancer = composeEnhancer(applyMiddleware(...middlewares));

const store = createStore(rootReducer(history), preLoadedState, enhancer);

store.subscribe(
  throttle(() => {
    const { auth } = store.getState();
    return saveState({ auth });
  }, 1000)
);
const configureStore = () => store;

export default configureStore;
