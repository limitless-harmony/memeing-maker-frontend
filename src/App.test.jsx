import React from 'react';
import { shallow } from 'enzyme';

import { App } from 'App';
import { history } from 'store';

describe('Testing the Root component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App history={history} />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
});
