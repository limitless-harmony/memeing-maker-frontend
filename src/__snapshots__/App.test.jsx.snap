// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Root component renders without crashing 1`] = `
<Connect(ConnectedRouterWithContext)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <styled.div>
    <GlobalStyleComponent />
    <Connect(Header) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/auth/:provider"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </styled.div>
</Connect(ConnectedRouterWithContext)>
`;
